lib/src/core/optional.dart

```diff
@@ -46,15 +46,15 @@ class Optional<T> extends IterableBase<T> {
   /// Gets the Optional value.
   ///
   /// Throws [StateError] if [value] is null.
-  T get value {
+  T/*!*/ get value {
     if (_value == null) {
       throw StateError('value called on absent Optional.');
     }
     return _value;
   }
 
   /// Executes a function if the Optional value is present.
-  void ifPresent(void ifPresent(T value)) {
+  void ifPresent(void ifPresent(T/*!*/ value)) {
```

Comment:
Here and the transform methods below -- we should annotate the function params as `/*!*/` but I need to look up how :)

Reply:
AFAICT you cannot use `T!`. You have to make T non nullable with `Optional<T extends Object>` and use `T?` where a nullable value can appear.

Reply:
Interesting. Alright will do some experimenting with this. Thanks @a14n.